tests/Browser/UserSearchTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Tests\Browser;
+
+use Tests\DuskTestCase;
+use Laravel\Dusk\Browser;
+use Tests\Browser\Pages\UserSearchPage;
+use Tests\Browser\Pages\HomePage;
+use Tests\Browser\Pages\UserPage;
+use Rogue\Models\Signup;
+
+class UserSearchTest extends DuskTestCase
+{
+    /**
+     * Test visiting the User Search Page.
+     * Searching for a user by name (instead of
+     * valid email address) returns "No user found!" message.
+     */
+    public function testIncorrectUserSearchPage()
+    {
+        // Create a signup so that the campaign over page will load.
+        $signup = factory(Signup::class)->create();
+
+        $this->browse(function (Browser $browser) {
+            $browser->visit(new HomePage)
+                    ->click('@login-button')
+                    ->assertPathIs('/register')
+                    ->clickLink('Log In')
+                    ->type('username', 'clee@dosomething.org')
+                    ->type('password', env('NORTHSTAR_PASSWORD'))
+                    ->press('Log In')
+                    ->assertPathIs('/campaigns')
+                    ->clickLink('User Search')
+                    ->assertPathIs('/users')
+                    ->on(new UserSearchPage)
+                    ->assertSeeIn('@title', 'Users')
+                    ->type('@search', 'taylor')
+                    ->press('Submit')
+                    ->assertSeeIn('@messages', 'No user found!');
+        });
+    }
+
+    /**
+     * Test visiting the User Search Page.
+     * Searching for a user by valid email
+     * redirects to the user's page.
+     */
+    public function testCorrectUserSearchPage()
+    {
+        // Create a signup so that the campaign over page will load.
+        $signup = factory(Signup::class)->create();
+
+        $this->browse(function (Browser $browser) {
+            $browser->visit(new HomePage)
+                    // We're already logged in from the first test.
+                    ->assertPathIs('/campaigns')
+                    ->clickLink('User Search')
+                    ->assertPathIs('/users')
+                    ->on(new UserSearchPage)
+                    ->assertSeeIn('@title', 'Users')
+                    ->type('@search', 'clee@dosomething.org')
+                    ->press('Submit')
+                    ->on(new UserPage)
+                    ->assertSeeIn('@title', 'User');
```

Comment:
If we decide to use a test user, we will know their NS id and then we can maybe do an assertion like `->assertPathIs'/users/:northstar_id`?

Reply:
I think asserting the page content is probably enough here – maybe look for the user's name in the heading?

Reply:
ok! I guess the only way we can tell what the user's name is without making a new call is to also add this in the `.env.dusk.local` file as a variable? Or, I can just make sure that the "User Info" heading is there? 

![screen shot 2017-11-16 at 10 54 23 am](https://user-images.githubusercontent.com/9019452/32900917-92ab8b16-cabc-11e7-8ae3-a8ce353946e9.png)

Reply:
oh wait, no we have the email address as an env variable so I can just use that!

Reply:
Ah yeah, smart! 👍